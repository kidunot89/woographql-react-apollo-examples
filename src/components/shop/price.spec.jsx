import React from 'react';
import { render } from '@testing-library/react';
import { toMatchDiffSnapshot } from 'snapshot-diff';

import Price from './price';

expect.extend({ toMatchDiffSnapshot });
describe('Price component', () => {
  it('renders simple product pricing successfully', () => {
    const { baseElement } = render(
      <Price
        regularPrice="$20.00"
        price="$20.00"
        type="SIMPLE"
      />
    );
    expect(baseElement).toMatchSnapshot();

    expect(
      <Price
        regularPrice="$20.00"
        price="$20.00"
        type="SIMPLE"
      />
    ).toMatchDiffSnapshot(
      <Price
        onSale
        regularPrice="$20.00"
        price="$18.99"
        type="SIMPLE"
      />
    );
  });

  it('renders variable product pricing successfully', () => {
    const { baseElement } = render(
      <Price
        regularPrice="$15.00-$20.00"
        price="$15.00-$20.00"
        type="VARIABLE"
      />
    );
    expect(baseElement).toMatchSnapshot();

    expect(
      <Price
        regularPrice="$15.00-$20.00"
        price="$15.00-$20.00"
        type="VARIABLE"
      />
    ).toMatchDiffSnapshot(
      <Price
        onSale
        regularPrice="$15.00-$20.00"
        price="$14.99-$18.99"
        type="VARIABLE"
      />
    );
  });
});
