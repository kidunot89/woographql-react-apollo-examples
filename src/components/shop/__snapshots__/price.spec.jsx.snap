// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component renders simple product pricing successfully 1`] = `
<body>
  <div>
    <p
      class="mt-0.5 mx-0.5 mb-0 text-sm text-shark group-hover:text-shark-900"
    >
      $20.00
    </p>
  </div>
</body>
`;

exports[`Price component renders simple product pricing successfully 2`] = `
"Snapshot Diff:
- <ProductPrice price=\\"$20.00\\" regularPrice=\\"$20.00\\" type=\\"SIMPLE\\" />
+ <ProductPrice onSale={true} price=\\"$18.99\\" regularPrice=\\"$20.00\\" type=\\"SIMPLE\\" />

  <p
    className=\\"mt-0.5 mx-0.5 mb-0 text-sm text-shark group-hover:text-shark-900\\"
  >
-   $20.00
+   <span
+     className=\\"line-through text-lynch group-hover:text-blue-bayoux\\"
+   >
+     $20.00
+   </span>
+    
+   $18.99
  </p>"
`;

exports[`Price component renders variable product pricing successfully 1`] = `
<body>
  <div>
    <p
      class="mt-0.5 mx-0.5 mb-0 text-sm text-shark group-hover:text-shark-900"
    >
      $15.00-$20.00
    </p>
  </div>
</body>
`;

exports[`Price component renders variable product pricing successfully 2`] = `
"Snapshot Diff:
- <ProductPrice price=\\"$15.00-$20.00\\" regularPrice=\\"$15.00-$20.00\\" type=\\"VARIABLE\\" />
+ <ProductPrice onSale={true} price=\\"$14.99-$18.99\\" regularPrice=\\"$15.00-$20.00\\" type=\\"VARIABLE\\" />

  <p
    className=\\"mt-0.5 mx-0.5 mb-0 text-sm text-shark group-hover:text-shark-900\\"
  >
-   $15.00-$20.00
+   $14.99-$18.99
  </p>"
`;
